docs/roles.md

```diff
@@ -0,0 +1,82 @@
+# Security Analysis HATVaults 2.0
+
+The contracts in this repository define a number of different roles that have the power to change the way the contracts work. We list them here, with a short description of what an account with this role can do, and to which account this role will be assigned on deployment
+
+## `HatVaults.ProxyOwner`
+
+- set to `HATTimelockController`
+- can upgrade the proxy
+- can `transferOwnership` and `renounceOwnership`
+
+## `HatVaults.owner`
+
+- set to `HATTimelockController`
+- can `transferOwnership` and `renounceOwnership` of `HATVaults`
+- can call `addPool` to add a pool
+- can call `setPool` to change pool properties
+- can call `approveClaim`
+- can call `swapBurnSend` and swap, burn, and send pool tokens that are earmarked for payout (after approveClaim)
+- can call `setClaimFee`
+- can call `setWithdrawRequestParams` (set time limits for withdrawal requests)
+- can call `setWithdrawSafetyPeriod` (set the times when the vault is locked for withdrawal)
+- can call `setBountySplit` (set how the bounty is split)
+- can call `setRouterWhitelistStatus` (add or remove address from the whitelist of routers)
+- can call `setVestingParams` (set vesting parameters for bounty paid in lptoken)
+- can call `setHatVestingParams` (set vesting paramaters for bounty paid in hats)
+- can call `setBountyLevelsDelay` (set how long a committee needs to wait bf changing the bounty levels)
+- can call `setFeeSetter` (set the feeSetter role)
+- can call `setPoolInitialized` (Set an "initialized" flag to disable "setShares")
+- can call `setShares` but only if the pool is not initialized
+- can call `setCommittee` but only if the committee has not checked in yet
+
+## `HatVaults - Vault committee` (committee set by each vault)
+
+- a specific address for each vault, typically a multisig
+- can call `committeeCheckIn`
+- can call `setCommittee`
+- can call `submitClaim`
+- can call `setPendingBountyLevels`
```

Comment:
setPendingMaxBounty -  Set pending request to set vault's max bounty that can be approved after pending period